File: bus/src/main/java/com/iwdael/livedatabus/Observable.java
package com.iwdael.livedatabus;

import androidx.annotation.NonNull;
import androidx.lifecycle.LifecycleOwner;
import androidx.lifecycle.Observer;

public interface Observable<T> {

    void observe(@NonNull LifecycleOwner owner, @NonNull Observer<? super T> observer);

    void observeSticky(@NonNull LifecycleOwner owner, @NonNull Observer<? super T> observer);

    void observeForever(@NonNull Observer<? super T> observer);

    void observeForeverSticky(@NonNull Observer<? super T> observer);

    void removeObserver(@NonNull Observer<? super T> observer);
}

File: core/src/main/java/com/iwdael/livedatabus/Observable.kt
package com.iwdael.livedatabus

import androidx.lifecycle.LifecycleOwner
import com.iwdael.livedatabus.ObservableWrapper
import androidx.lifecycle.MutableBusLiveData
import androidx.lifecycle.Observer
/**
 * author : Iwdael
 * e-mail : iwdael@outlook.com
 */
interface Observable<T> {
    fun observe(owner: LifecycleOwner, observer: Observer<in T>)
    fun observeSticky(owner: LifecycleOwner, observer: Observer<in T>)
    fun observeForever(observer: Observer<in T>)
    fun observeForeverSticky(observer: Observer<in T>)
    fun removeObserver(observer: Observer<in T>)
}